eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/Tick.hs

```diff
@@ -167,6 +175,63 @@ validatingTickTransition nes slot = do
 
   pure $ nes' {nesEs = es''}
 
+-- | This is a limited version of 'validatingTickTransition' which is only suitable
+-- for the future ledger view.
+validatingTickTransitionFORECAST ::
+  forall tick era.
+  ( State (tick era) ~ NewEpochState era,
+    BaseM (tick era) ~ ShelleyBase,
+    State (EraRule "UPEC" era) ~ UpecState era,
+    Signal (EraRule "UPEC" era) ~ (),
+    Environment (EraRule "UPEC" era) ~ EpochState era,
+    Embed (EraRule "UPEC" era) (tick era),
+    STS (tick era)
+  ) =>
+  NewEpochState era ->
+  SlotNo ->
+  TransitionRule (tick era)
+validatingTickTransitionFORECAST nes slot = do
+  -- This whole function is a specialization of an inlined 'NEWEPOCH'.
+  --
+  -- The ledger view, 'LedgerView', is built entirely from the 'nesPd' and 'esPp' and
+  -- 'dsGenDelegs', so the correctness of 'validatingTickTransitionFORECAST' only
+  -- depends on getting these three fields correct.
+
+  epoch <- liftSTS $ do
+    ei <- asks epochInfoPure
+    epochInfoEpoch ei slot
+
+  let es = nesEs nes
+      ss = esSnapshots es
+
+  -- the relevant 'NEWEPOCH' logic
+  let pd' = ssStakeMarkPoolDistr ss
+
+  -- note that the genesis delegates are updated not only on the epoch boundary.
+  if unEpochNo epoch /= unEpochNo (nesEL nes) + 1
+    then pure $ nes {nesEs = adoptGenesisDelegs es slot}
+    else do
+      -- We can skip 'SNAP'; we already have the equivalent pd'.
+
+      -- We can skip 'MIR' and 'POOLREAP';
+      -- we don't need to do the checks:
+      -- if the checks would fail, then the node will fail in the 'TICK' rule
+      -- if it ever then node tries to validate blocks for which the
+      -- return value here was used to validate their headers.
```

Comment:
I found it hard to read this two lines. It feels like some words may be missing.

Reply:
yea, it's not great. how does this sound:

```
We can skip 'MIR' and 'POOLREAP'.
We do not need to do the checks in the 'TICK' rule since they are not actually
validating user input, but are instead a protection for the internal consistency
of the ledger state.
```

Reply:
Yes, that sounds better! :+1: